src/views/Users/components/UserCard/UserRoleOrPathwayChips.js

```diff
@@ -0,0 +1,288 @@
+import React,{useEffect} from "react"
```

Comment:
This comment is relevant for the whole file and not this line alone.

Here you are essentially not reusing a lot of code by combining the functionality with UserRoleChips and PathwayChips. Overall there can be different files for PathwayChips and UserRoleChips. But there can be a lot of code that can be reused within both of these components.

Example: *Dialog box and the text box over there is common. The functionality when you click on save and cancel on the dialog box also remains almost the same only the API request changes.*

Maybe you can make a <ChipListWithUpdate/> which can then be given a few attributes. 

1. `optionList`
2. `onRemove` - this will be called when a chip within a list is removed. it can be passed some specific parameters and that way you can do your own custom stuff in role chips v/s pathway chips.
3. `onAdd`
4. `getAddSelectOptions` - this will get options to populate the select in the dropdown. for pathways it can just return a list while for pathways it can do an API call. 

Once you make it like this, it can  be used at different places too when needed.

Reply:
I created bhaiyya you need to check that, I am pushing changes